docs/en/advanced/lazy-loading.md

```diff
@@ -45,3 +45,52 @@ const Baz = () => import(/* webpackChunkName: "group-foo" */ './Baz.vue')
 ```
 
 webpack will group any async module with the same chunk name into the same async chunk.
+
+### Handling Loading State
+
+Async component factories can also [return an object](https://vuejs.org/v2/guide/components-dynamic-async.html#Handling-Loading-State), specifying alternative components to use when loading or if an error occurs. This is a great way to improve user experience on slow or intermittent connections.
+
+Unfortunately, routes can currently only resolve to a single component in Vue Router. So to make this work, you'll need a helper function to create an intermediary component, like this:
+
+```js
+function lazyLoadView (AsyncView) {
+  const AsyncHandler = () => ({
+    component: AsyncView,
+    // A component to use while the component is loading.
+    loading: require('./Loading.vue').default,
+    // A fallback component in case the timeout is exceeded
+    // when loading the component.
+    error: require('./Timeout.vue').default,
+    // Delay before showing the loading component.
+    // Default: 200 (milliseconds).
+    delay: 400,
+    // Time before giving up trying to load the component.
+    // Default: Infinity (milliseconds).
+    timeout: 10000
+  })
+
+  return Promise.resolve({
```

Comment:
Doesn't it work without returning a promise and just doing `component: lazyLoadView(...)`

Reply:
If the `import(...)` isn't behind a function to be evaluated later, won't all the chunks be downloaded at startup time, regardless of whether the user needs them?

Reply:
As what I know only on execution its fetched (always forget how it works)

Reply:
So ies needs to be in a function, so it is only fetched when needed